octopusdeploy/config.go

```diff
@@ -40,18 +44,37 @@ func (c *Config) Client() (*octopusdeploy.Client, diag.Diagnostics) {
 	}
 
 	if len(c.SpaceName) > 0 {
+		log.Printf("[DEBUG] Config is locating space using name '%s'", c.SpaceName)
 		spaces, err := client.Spaces.Get(octopusdeploy.SpacesQuery{
-			Name: c.SpaceName,
+			PartialName: c.SpaceName,
 		})
 		if err != nil {
 			return nil, diag.FromErr(err)
 		}
 
-		client, err = octopusdeploy.NewClient(nil, apiURL, c.APIKey, spaces.Items[0].GetID())
+		if spaces.TotalResults == 0 {
+			return nil, diag.Errorf("Unable to locate space with name '%s', found no spaces", c.SpaceName)
+		}
+		if spaces.TotalResults > 1 {
```

Comment:
I think we should remove this check and instead try and find an exact match of `c.SpaceName` within `spaces.TotalResults`. If we don't find an exact match, we should error despite the size of the array (even if we have only 1 item in the array).

This is to prevent confusion with the user accidentally matching the wrong space or unable to find a space that has a name that overlaps (example: `space1` & `space10`)